.test-infra/jenkins/job_PerformanceTests_KafkaIO_IT.groovy

```diff
@@ -21,23 +21,37 @@ import Kubernetes
 import InfluxDBCredentialsHelper
 
 String jobName = "beam_PerformanceTests_Kafka_IO"
+String HIGH_RANGE_PORT = "32767"
 
 job(jobName) {
-  common.setTopLevelMainJobProperties(delegate)
-  // TODO(https://github.com/apache/beam/issues/20333): Re-enable once fixed.
-  // common.setAutoJob(delegate, 'H H/6 * * *')
-  //  common.enablePhraseTriggeringFromPullRequest(
-  //      delegate,
-  //      'Java KafkaIO Performance Test',
-  //      'Run Java KafkaIO Performance Test')
+  common.setTopLevelMainJobProperties(delegate, 'master', 120)
+  common.setAutoJob(delegate, 'H H/6 * * *')
+  common.enablePhraseTriggeringFromPullRequest(
+      delegate,
+      'Java KafkaIO Performance Test',
+      'Run Java KafkaIO Performance Test')
   InfluxDBCredentialsHelper.useCredentials(delegate)
 
   String namespace = common.getKubernetesNamespace(jobName)
   String kubeconfig = common.getKubeconfigLocationForNamespace(namespace)
   Kubernetes k8s = Kubernetes.create(delegate, kubeconfig, namespace)
-  k8s.apply(common.makePathAbsolute("src/.test-infra/kubernetes/kafka-cluster"))
 
+  String kafkaDir = common.makePathAbsolute("src/.test-infra/kubernetes/kafka-cluster")
+  String kafkaTopicJob="job.batch/kafka-config-eff079ec"
+
+  // Select available ports for services and avoid collisions
+  steps {
+    String[] configuredPorts = ["32400", "32401", "32402"]
+    (0..2).each { service ->
+      k8s.availablePort(service == 0 ? configuredPorts[service] : "\$KAFKA_SERVICE_PORT_${service-1}",
+          HIGH_RANGE_PORT, "KAFKA_SERVICE_PORT_$service")
+      shell("sed -i -e s/${configuredPorts[service]}/\$KAFKA_SERVICE_PORT_$service/ \
+                  ${kafkaDir}/04-outside-services/outside-${service}.yml")
+    }
+  }
```

Comment:
Can you add a comment so I can understand what this does? To be honest, I don't know where `KAFKA_SERVICE_PORT_` and other special variables are coming from.
Please also document this in the confluence wiki.

Reply:
Please add a comment at the top describing what resources are used by this test and what the test does.

Reply:
Hi @pablo, I added some comments for describing the requested parts in the code.